core/node/api_server/src/execution_sandbox/execute.rs

```diff
@@ -1,80 +1,80 @@
 //! Implementation of "executing" methods, e.g. `eth_call`.
 
-use anyhow::Context as _;
-use tracing::{span, Level};
-use zksync_dal::{ConnectionPool, Core};
-use zksync_multivm::{
-    interface::{
-        TransactionExecutionMetrics, TxExecutionMode, VmExecutionResultAndLogs, VmInterface,
-    },
-    tracers::StorageInvocations,
-    MultiVMTracer,
+use async_trait::async_trait;
+use zksync_dal::{Connection, Core};
+use zksync_multivm::interface::{
+    storage::ReadStorage, BytecodeCompressionError, OneshotEnv, TransactionExecutionMetrics,
+    VmExecutionResultAndLogs,
 };
 use zksync_types::{
-    l2::L2Tx, transaction_request::CallOverrides, ExecuteTransactionCommon, Nonce,
+    api::state_override::StateOverride, l2::L2Tx, ExecuteTransactionCommon, Nonce,
     PackedEthSignature, Transaction, U256,
 };
 
 use super::{
-    apply, testonly::MockTransactionExecutor, vm_metrics, ApiTracer, BlockArgs, TxSharedArgs,
-    VmPermit,
+    apply::{self, MainOneshotExecutor},
+    storage::StorageWithOverrides,
+    testonly::MockOneshotExecutor,
+    vm_metrics, ApiTracer, BlockArgs, OneshotExecutor, TxSetupArgs, VmPermit,
 };
-use crate::execution_sandbox::api::state_override::StateOverride;
 
+/// Executor-independent arguments necessary to for oneshot transaction execution.
+///
+/// # Developer guidelines
+///
+/// Please don't add fields that duplicate `SystemEnv` or `L1BatchEnv` information, since both of these
+/// are also provided to an executor.
 #[derive(Debug)]
 pub(crate) struct TxExecutionArgs {
```

Comment:
Not 100% sure this grouping makes sense, but it sort of does IMO; these params are related specifically to oneshot execution.